torch/autograd/gradcheck.py

```diff
@@ -867,36 +870,46 @@ def fast_gradcheck(func, func_out, tupled_inputs, outputs, eps, rtol,
 
     # Use our own generator to avoid messing with the user's RNG state
     g_cpu = torch.Generator()
-    all_u = [vec_from_tensor(inp, g_cpu) for inp in inp_tensors]
+    all_ur = [vec_from_tensor(inp, g_cpu, True) for inp in inp_tensors]
+    all_ui = [vec_from_tensor(inp, g_cpu, True) for inp in inp_tensors]
```

Comment:
Does this mean that you always generate the ui, even for real inputs and just end up never using it?

Reply:
Updated to only generate when necessary.